test/e2e/e2e.go

```diff
@@ -2036,6 +2094,126 @@ var _ = ginkgo.Describe("e2e ingress traffic validation", func() {
 	})
 })
 
+var _ = ginkgo.Describe("e2e ingress to host-networked pods traffic validation", func() {
```

Comment:
describe from where the traffic is coming from

Reply:
can we also add test coverage for all of these cases please:
1. from external network -> external traffic policy local -> host network endpoint
2. from external network -> external traffic policy local -> ovn endpoint
3. from host -> external traffic policy local -> host network endpoint
4. from host -> external traffic policy local -> ovn endpoint
5. from ovn pod -> external traffic policy local -> host network endpoint
6. from ovn pod -> external traffic policy local -> ovn network pod

If we want to say we dont handle scenarios 5 and 6, I can agree with that. But we should document it in that markdown doc I asked for in another comment.

Reply:
Working on this

Reply:
This PR Has 1.and 2. which are the most relevant for this feature, I will continue to add the rest of these cases but I don't think it should block this PR.

Reply:
Also small Nit but I have a comment for here ^ that I will add after A final review

Reply:
Added a comment for this test above the test Context